pkg/pushsync/pushsync.go

```diff
@@ -147,10 +167,86 @@ func (ps *PushSync) handler(ctx context.Context, p p2p.Peer, stream p2p.Stream)
 				return fmt.Errorf("chunk store: %w", err)
 			}
 
+			count := 0
+			// Push the chunk to some peers in the neighborhood in parallel for replication.
+			// Any errors here should NOT impact the rest of the handler.
+			err = ps.topologyDriver.EachNeighbor(func(peer swarm.Address, po uint8) (bool, bool, error) {
+
+				// skip forwarding peer
+				if peer.Equal(p.Address) {
+					return false, false, nil
+				}
+
+				if count == nPeersToPushsync {
+					return true, false, nil
+				}
+				count++
+
+				go func(peer swarm.Address) {
+
+					var err error
+					defer func() {
+						if err != nil {
+							ps.logger.Tracef("pushsync replication: %v", err)
+						}
+					}()
+
+					// price for neighborhood replication
+					const receiptPrice uint64 = 0
+
+					headers, err := headerutils.MakePricingHeaders(receiptPrice, chunk.Address())
+					if err != nil {
+						err = fmt.Errorf("make pricing headers: %w", err)
+						return
+					}
+
+					streamer, err := ps.streamer.NewStream(ctx, peer, headers, protocolName, protocolVersion, streamName)
+					if err != nil {
+						err = fmt.Errorf("new stream for peer %s: %w", peer.String(), err)
+						return
+					}
+					defer streamer.Close()
+
+					returnedHeaders := streamer.Headers()
+					_, returnedPrice, returnedIndex, err := headerutils.ParsePricingResponseHeaders(returnedHeaders)
+					if err != nil {
+						err = fmt.Errorf("push price headers read returned: %w", err)
+						return
+					}
+
+					// check if returned price matches presumed price, if not, return early.
+					if returnedPrice != receiptPrice {
+						err = ps.pricer.NotifyPeerPrice(peer, returnedPrice, returnedIndex) // save priceHeaders["price"] corresponding row for peer
```

Comment:
dunno if the comment here is needed as it leaks internals of what happens in the pricer